src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -126,6 +145,36 @@ public final void onRunFinish(boolean succeeded, BatchSinkContext context) {
     }
   }
 
+  private void recordMetric(BatchSinkContext context, Job queryJob) throws Exception {
+    String operation = baseConfiguration.get(BigQueryConstants.CONFIG_OPERATION);
+    String metricName = String.format("records.%s", operation.toLowerCase());
+    long totalRows = getTotalRows(queryJob);
```

Comment:
is this the only thing that can throw an exception? It would be cleaner to just have this return without emitting any metric rather than have it throw an exception for an unknown query type and make the caller handle an exception.

Reply:
Done